### PR TITLE
Fix NumPy array creation error by specifying object type

### LINKED ISSUE #1065: PySMO: Fix broken np.array calls due to numpy changes
Changes to numpy have broken some numpy calls in PySMO. Calls of the type:

``t1 = np.array([list of variables])`` 

which worked fine as at numpy 1.22.3 no longer work in some numpy 1.24.1 cases. 

The failures appears to occur trying converting a list of a mixture of variable types into an array e.g.,  lists containing mixtures of scalar and indexed pyomo variables.

### PR DESCRIPTION
**Pull request by OOAmusat**
## Fixes
#1065 

## Summary/Motivation:
Fixes potential NumPy errors in surrogate model expression generation when inputs are of mixed pyomo types, e.g scalar and indexed variables. 

## Changes proposed in this PR:
- specify ``dtype=object`` in PySMO when creating Numpy arrays from a list of pyomo variables.
- Add test for mixed variable input types.

### Legal Acknowledgement

By contributing to this software project, I agree to the following terms and conditions for my contribution:

1. I agree my contributions are submitted under the license terms described in the LICENSE.txt file at the top level of this directory.
2. I represent I am authorized to make the contributions and grant the license. If my employer has rights to intellectual property that includes these contributions, I represent that I have received permission to make contributions and grant the required license on behalf of that employer.

### CODE REVIEW

File: idaes/core/surrogate/tests/test_pysmo_surrogate.py [hunk 1]
```diff
@@ -1633,6 +1633,22 @@ def test_populate_block_multisurrogate_poly_userdef(self, pysmo_surr4):
...
+
+        m = ConcreteModel()
+        m.x1 = Var(initialize=0, bounds=(0, 5))
+        m.x2 = Var([0], initialize=0, bounds=(0, 10))
+        m.z1 = Var(initialize=0, bounds=(0, 10))
+        m.surrogate = SurrogateBlock(concrete=True)
+        m.surrogate.build_model(
+            pysmo_surr1, input_vars=[m.x1, m.x2], output_vars=[m.z1]
+        )
+        print(m)
```
Comment (andrewlee94) [inline idaes/core/surrogate/tests/test_pysmo_surrogate.py:1648]: The `print` and `display` lines should probably be removed from the test.

**Review by andrewlee94 — APPROVED** [review]
One minor comment about some print statements in the tests but otherwise look good to me.

File: idaes/core/surrogate/tests/test_pysmo_surrogate.py (hunk 1 above)
Comment (OOAmusat) [reply idaes/core/surrogate/tests/test_pysmo_surrogate.py]: That was an oversight and has been fixed.

**Review by ksbeattie — APPROVED** [review]
LGTM

**Pull request merged**